ue4docker/infrastructure/BuildConfiguration.py

```diff
@@ -226,15 +238,24 @@ def _generateWindowsConfig(self):
 		hostSystem32 = os.path.join(os.environ['SystemRoot'], 'System32')
 		self.defaultDllDir = hostSysnative if os.path.exists(hostSysnative) else hostSystem32
 		self.dlldir = self.args.dlldir if self.args.dlldir is not None else self.defaultDllDir
-		
+
+		self.visualStudio = self.args.visual_studio
+
+		# Check whether specified Unreal Engine release is compatible with specified Visual Studio
+		vs_min_supported_unreal = VisualStudio.MinSupportedUnreal.get(self.visualStudio, None)
+		if vs_min_supported_unreal is not None and self.release < vs_min_supported_unreal:
+			raise RuntimeError('specified version of Unreal Engine cannot be built with Visual Studio 2019, oldest supported is {}'.format(vs_min_supported_unreal))
```

Comment:
"2019" here probably should be `self.visualStudio`, so that the error message doesn't have to change later.

Reply:
Right, I've overlooked this place when was refactoring things around.

Reply:
Fixed